file/sst_file_manager_impl.cc

```diff
@@ -440,24 +442,15 @@ void SstFileManagerImpl::WaitForEmptyTrash() {
 }
 
 void SstFileManagerImpl::OnAddFileImpl(const std::string& file_path,
-                                       uint64_t file_size, bool compaction) {
+                                       uint64_t file_size) {
   auto tracked_file = tracked_files_.find(file_path);
   if (tracked_file != tracked_files_.end()) {
     // File was added before, we will just update the size
-    assert(!compaction);
     total_files_size_ -= tracked_file->second;
     total_files_size_ += file_size;
     cur_compactions_reserved_size_ -= file_size;
   } else {
     total_files_size_ += file_size;
-    if (compaction) {
-      // Keep track of the size of files created by in-progress compactions.
-      // When calculating whether there's enough headroom for new compactions,
-      // this will be subtracted from cur_compactions_reserved_size_.
-      // Otherwise, compactions will be double counted.
-      in_progress_files_size_ += file_size;
-      in_progress_files_.insert(file_path);
-    }
```

Comment:
Unless I'm missing something, I think we could remove `in_progress_files_` and `in_progress_files_size_` altogether from this class (i.e. even the member variables, as well as any code that touches them). I believe this was the only place where files could get added to these structures, so if this is dead code (due to `compaction` always being `false`), so is any other chunk of code that queries the map or deletes entries etc.

Reply:
```in_progress_files_size_``` is used during compaction to check if there is enough headroom for compaction by db_impl_compaction_flush.cc. So I couldn't remove it at other places. https://github.com/facebook/rocksdb/blob/1a343bc393b5328c9825896a20fc750fb4de3e1c/file/sst_file_manager_impl.cc#L202

Reply:
But since in_progress_files_size_ is always zero, i guess we can remove this. Let me check again.

Reply:
> `in_progress_files_size_` is used during compaction to check if there is enough headroom for compaction by db_impl_compaction_flush.cc. So I couldn't remove it at other places.

The way I read the code, `in_progress_files_size_` is always zero, since it could only be incremented if someone called `OnAddFile` with `compaction==true`, which doesn't happen, right?

If so, any code like `needed_headroom -= in_progress_files_size_` is a no-op.

Reply:
P.S. I'm bringing this up because if for some reason it was necessary to track the "in progress" files of compactions, we would need to _extend_ this logic to cover blobs. But hopefully, we can just ditch it altogether :)

Reply:
Yes, you are right. It will always be zero and we won't add anything to the map so we can remove the variables altogether  from the code. I will update the code.

Reply:
> But since in_progress_files_size_ is always zero, i guess we can remove this. Let me check again.

Yep, and `in_progress_files_` is supposed to be always empty, so any lookups in it would be unsuccessful.